### PR TITLE
Update module to use new testing approach

### LINKED ISSUE #15: Update testing to new approach
We should update the testing to use Cloud Build and a new Service Account per run, as documented [here](https://docs.google.com/document/d/1eiHpZlhEDMDG8gi4LHyv2aZs1TJmeFlF8sqCE57_j4A/edit#heading=h.rz8l5qn0ygc0).

### PR DESCRIPTION
**Pull request by paulpalamarchuk**
* Fix #15
 * Linters fixed
 * Workaround for kitchen verify (issue #21)
 * Moved resource naming to variables in `simple_example`

### CODE REVIEW

File: examples/onprem_demo/README.md [hunk 1]
```diff
@@ -6,13 +6,13 @@ This is a demo of a GCP project protected from internet access using [VPC Servic
 - Creates a project that will be protected with VPC SC with a custom network that has only one `10.7.0.0/16` subnet (purely for the sake of simplicity) in a region of your choice
 - Creates a VPN tunnel connecting the two projects (this could conceptually represent a dedicated interconnect as well)
 - Creates a Cloud Router in each project to exchange network routes between the two environments
-- Launches one virtual machine in the VPC SC project: 
+- Launches one virtual machine in the VPC SC project:
     - one Windows instance without an external IP address
-- Launches two virtual machines in the on-prem project: 
-    - one Linux instance acting as a forward proxy for all internet-bound requests; and 
+- Launches two virtual machines in the on-prem project:
+    - one Linux instance acting as a forward proxy for all internet-bound requests; and
```
Comment (kopachevsky) [inline examples/onprem_demo/README.md]: what this "and" means?

Comment (paulpalamarchuk) [reply examples/onprem_demo/README.md]: As I understand the meaning: ` Launches two virtual machines in the on-prem project: one Linux instance ... ; and one Windows ...`

**Comment by paulpalamarchuk** [comment]
Tests:
image

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+
+## Testing
+
+### Requirements
+- [bundler](https://github.com/bundler/bundler)
+- [gcloud](https://cloud.google.com/sdk/install)
+- [terraform-docs](https://github.com/segmentio/terraform-docs/releases) 0.3.0
+
+### Autogeneration of documentation from .tf files
```
Comment (morgante) [inline CONTRIBUTING.md]: Why is this section header repeated?

Clearly this PR did not include attention to detail.

File: test/fixtures/shared/variables.tf [hunk 3]
```diff
@@ -43,3 +43,9 @@ variable "members" {
   description = "An allowed list of members (users, service accounts). The signed-in identity originating the request must be a part of one of the provided members. If not specified, a request may come from any user (logged in/not logged in, etc.). Formats: user:{emailid}, serviceAccount:{emailid}"
   type        = list(string)
 }
+
+variable "random_suffix" {
```
Comment (morgante) [inline test/fixtures/shared/variables.tf]: There is 0 reason for this to be a variable.

@paulpalamarchuk Just make this a random hex like we do everywhere else.

File: test/helper_gcloud_auth.sh [hunk 4]
```diff
@@ -12,8 +14,6 @@
...
 # limitations under the License.
 
-ruby '2.5.3'
-
-source 'https://rubygems.org/' do
-  gem 'kitchen-terraform', '~> 4.0.3'
-end
+# Login to GCP for using bq-script (issue #21)
+# shellcheck disable=SC2154
+gcloud auth activate-service-account "${SA_EMAIL}" --key-file="${GOOGLE_APPLICATION_CREDENTIALS}" --project="${TF_VAR_project_id}"
```
[the remark this answers is not in the document]
Comment (paulpalamarchuk) [reply test/helper_gcloud_auth.sh]: @morgante I've created 2 PRs:
[Add SA activation to init_credentials function](https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/pull/368)
[Add function to remove any existing Access Policy](https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/pull/369)

File: test/fixtures/shared/variables.tf (hunk 3 above)
Comment (paulpalamarchuk) [reply test/fixtures/shared/variables.tf]: Moved inside `simple_example` fixture

File: .dockerignore [hunk 5]
```diff
@@ -0,0 +1,10 @@
+.git
+.terraform
+.terraform.d
+.kitchen
+terraform.tfstate.d
+test/fixtures/*/.terraform
```
[the remark this answers is not in the document]
Comment (paulpalamarchuk) [reply .dockerignore]: removed

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+
+### Linting
+The makefile in this project will lint or sometimes just format any shell,
+Python, golang, Terraform, or Dockerfiles. The linters will only be run if
+the makefile finds files with the appropriate file extension.
+
+All of the linter checks are in the default make target, so you just have to
+run
+
+```
```
[the remark this answers is not in the document]
Comment (paulpalamarchuk) [reply CONTRIBUTING.md]: removed

File: CONTRIBUTING.md (hunk 2 above)
Comment (paulpalamarchuk) [reply CONTRIBUTING.md]: removed

File: build/int.cloudbuild.yaml [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  args: ['/bin/bash', '-c', 'source /usr/local/bin/task_helper_functions.sh && source_test_env && init_credentials && remove_gcloud_org_accesspolicy']
+- id: create
+  name: 'gcr.io/cloud-foundation-cicd/$_DOCKER_IMAGE_DEVELOPER_TOOLS:$_DOCKER_TAG_VERSION_DEVELOPER_TOOLS'
+  args: ['/bin/bash', '-c', 'source /usr/local/bin/task_helper_functions.sh && kitchen_do create']
+- id: converge
+  name: 'gcr.io/cloud-foundation-cicd/$_DOCKER_IMAGE_DEVELOPER_TOOLS:$_DOCKER_TAG_VERSION_DEVELOPER_TOOLS'
+  args: ['/bin/bash', '-c', 'source /usr/local/bin/task_helper_functions.sh && kitchen_do converge']
+- id: verify
+  name: 'gcr.io/cloud-foundation-cicd/$_DOCKER_IMAGE_DEVELOPER_TOOLS:$_DOCKER_TAG_VERSION_DEVELOPER_TOOLS'
+  args: ['/bin/bash', '-c', 'source /usr/local/bin/task_helper_functions.sh && source_test_env && init_credentials && activate_gcloud_sa && sleep 360 && kitchen_do verify']
```
Comment (morgante) [inline build/int.cloudbuild.yaml]: We shouldn't need to do all this, especially now that activate_gcloud_sa has been upstreamed. Please revise.

File: build/int.cloudbuild.yaml [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+- id: prepare
+  name: 'gcr.io/cloud-foundation-cicd/$_DOCKER_IMAGE_DEVELOPER_TOOLS:$_DOCKER_TAG_VERSION_DEVELOPER_TOOLS'
+  args: ['/bin/bash', '-c', 'source /usr/local/bin/task_helper_functions.sh && prepare_environment']
+  env:
+  - 'TF_VAR_org_id=$_ORG_ID'
+  - 'TF_VAR_folder_id=$_FOLDER_ID'
+  - 'TF_VAR_billing_account=$_BILLING_ACCOUNT'
+- id: remove existing policy
+  name: 'gcr.io/cloud-foundation-cicd/$_DOCKER_IMAGE_DEVELOPER_TOOLS:$_DOCKER_TAG_VERSION_DEVELOPER_TOOLS'
+  args: ['/bin/bash', '-c', 'source /usr/local/bin/task_helper_functions.sh && source_test_env && init_credentials && remove_gcloud_org_accesspolicy']
```
Comment (morgante) [inline build/int.cloudbuild.yaml]: Just include init_credentials/source_test_env within remove_gcloud_org_accesspolicy.

File: test/setup/main.tf [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+    "storage-api.googleapis.com",
+    "serviceusage.googleapis.com",
+    "accesscontextmanager.googleapis.com",
+    "bigquery-json.googleapis.com",
+    "bigquerystorage.googleapis.com",
+    "compute.googleapis.com"
+  ]
+}
+
+resource "google_project" "ci-project-policy-test" {
```
Comment (morgante) [inline test/setup/main.tf]: Please use project factory.

File: build/int.cloudbuild.yaml (hunk 7 above)
Comment (paulpalamarchuk) [reply build/int.cloudbuild.yaml]: Updated Docker image tag.

File: build/int.cloudbuild.yaml (hunk 8 above)
Comment (paulpalamarchuk) [reply build/int.cloudbuild.yaml]: Moved inside `remove_gcloud_org_accesspolicy` function

File: test/setup/main.tf (hunk 9 above)
Comment (paulpalamarchuk) [reply test/setup/main.tf]: Updated.

**Pull request merged**